novawallet/Common/ViewController/CardLayoutPresentationController/CardLayoutPresentationContoroller.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+final class CardLayoutPresentationController: UIViewController {
+    // we keep strong reference to delegate pan gesture processing
+    // because translation change is tied with transition.
+    // Delegate doesn't retains controller so we're safe here
+
+    // swiftlint:disable weak_delegate
+    private let transitionDelegate: CardLayoutTransitionDelegateProtocol
+
+    init(transitionDelegate: CardLayoutTransitionDelegateProtocol) {
+        self.transitionDelegate = transitionDelegate
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupActions()
+    }
+
+    static func topOffset() -> CGFloat {
+        (UIApplication.shared.keyWindow?.safeAreaInsets.top ?? 0) + 12
+    }
+}
+
+// MARK: Private
+
+private extension CardLayoutPresentationController {
+    func setupActions() {
+        let pan = UIPanGestureRecognizer(
+            target: self,
+            action: #selector(panGesture(_:))
+        )
+
+        view.addGestureRecognizer(pan)
+    }
+
+    @objc func panGesture(_ sender: UIPanGestureRecognizer) {
+        transitionDelegate.didReceivePanState(
```

Comment:
Did you check that this pan doesn't conflict with scrolling to the content view?

Reply:
Yes, no conflicts here